pkgs/by-name/te/telegram-macos/package.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenvNoCC,
+  unzip,
+  fetchurl,
+}:
+let
+  version = "10.15.2-265540";
+  versionTokens = lib.split "-" version;
+  versionFirst = lib.head versionTokens;
+  versionSecond = lib.last versionTokens;
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit version;
+
+  pname = "telegram-macos";
+
+  src = fetchurl {
+    url = "https://osx.telegram.org/updates/Telegram-${versionFirst}.${versionSecond}.app.zip";
+    hash = "sha256-ne087xxFihsv+3UMFFk3zGzaweRG9o8mVwXkMIwH/9Y=";
+  };
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [ unzip ];
+
+  installPhase = ''
```

Comment:
Is there a reason to not use `telegram-desktop`? [derivation](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/default.nix#L223) - includes support for Darwin already

Reply:
[`telegram-desktop`][tdesktop] is a completely different application. It is cross-platform, written in C++ with Qt, while the [one in question][telegramswift] is macOS-exlusive and implemented in Swift. 

[tdesktop]: https://github.com/telegramdesktop/tdesktop
[telegramswift]: https://github.com/overtake/TelegramSwift/tree/master

Reply:
Then perhaps it should be named `telegram-swift` per the repo and could be built from source.

Reply:
@JohnRTitor requires Xcode unfortunately, it's not just a Swift package - also, `telegram-swift` would be ambiguous, i.e. one may think it may also run on Linux.

Reply:
Is there a naming conflict with other projects named `telegram-swift` that do work on Linux? Otherwise I'd say just go with the upstream name, the `meta` attribute should indicate Linux incompatibility.